### PR TITLE
Improvements and refactoring

### PR DESCRIPTION
**Pull request by sompylasar**

### CODE REVIEW

**Comment by znrg** [comment]
+1

File: src/c5t.js [hunk 1]
```diff
@@ -1,47 +1,171 @@
...
+  var C5T_EVENT_ACTION_EXIT = 'Ex';
+  var C5T_EVENT_ACTION_FOREGROUND = 'Fg';
+  var C5T_EVENT_ACTION_BACKGROUND = 'Bg';
+  
+  // TODO(sompylasar): Update the API endpoint URL when the backend is ready.
+  var C5T_DATADROP_URL = '//localhost:4000/datadrop/v1/';
   
+  /**
+   * The internal name of the default `tracker` object.
+   * Not intended for using by the user's code.
```
Comment (dkorolev) [inline src/c5t.js:65]: *not intended for use by ...

File: src/c5t.js [hunk 2]
```diff
@@ -123,137 +257,400 @@
...
       navigator.sendBeacon(baseUrl, payloadString);
       setTimeout(callbackFn, 0);
     };
   }
   
+  function _serializeValue(value) {
+    return (
+      value === true ? "1" :
+      value === false ? "0" :
+      void 0 == value ? "" :
```
Comment (dkorolev) [inline src/c5t.js:288]: I couldn't understand this black magic. :(

**Comment by dkorolev** [comment]
Looks Good To Me!

File: src/c5t.js (hunk 2 above)
Comment (sompylasar) [inline src/c5t.js:288]: That's just shortening (mapping certain values to their shorter equivalents).

Comment (sompylasar) [inline src/c5t.js:288]: `void 0` returns the true `undefined`; `undefined == value` evaluates to true if `value` is `undefined` or `null`.
Honestly, I've copied this expression from GA.

**Pull request merged**